src/schemas/json/template.json

```diff
@@ -137,13 +146,103 @@
-    "generatorOther": {
+    "generatorCoalesce": {
       "properties": {
-        "generator": {
-          "enum": [ "casing", "coalesce", "port", "switch" ]
-        },
+        "generator": { "enum": [ "coalesce" ] },
+        "parameters": {
+          "type": "object",
+          "required": [ "sourceVariableName", "fallbackVariableName" ],
+          "properties": {
+            "sourceVariableName": {
+              "type": "string",
+              "description": "The name of the symbol whose value will be inspected - if the value is effectively equal to the default value, the value of the symbol referred to by fallbackVariableName is used, otherwise the value of this symbol"
+            },
+            "defaultValue": {
+              "type": "string",
+              "description": "The value to consider as being the default - if the value of the symbol referred to by sourceVariableName is equal to this, the value of the symbol referred to in fallbackVariableName is used instead",
+              "default": ""
+            },
+            "fallbackVariableName": {
+              "type": "string",
+              "description": "The name of the symbol to return the value of if the symbol referred to by sourceVariableName has a value equal to the value of defaultValue"
+            }
+          }
+        }
+      }
+    },
+    "generatorCasing": {
+      "properties": {
+        "generator": { "enum": [ "casing" ] },
+        "parameters": {
+          "required": [ "source" ],
+          "properties": {
+            "source": {
+              "description": "The name of the symbol whose value should have its case changed",
+              "type": "string"
+            },
+            "toLower": {
+              "description": "Whether the case changed to should be lower case, if false, the value is made to be uppercase instead",
+              "type": "boolean",
+              "default": true
+            }
+          }
+        }
+      }
+    },
+    "generatorPort": {
+      "properties": {
+        "generator": { "enum": [ "port" ] },
+        "parameters": {
+          "properties": {
+            "low": {
+              "description": "The lower bound of acceptable port numbers",
+              "type": "integer",
+              "default": 1024
+            },
+            "high": {
+              "description": "The upper bound of acceptable port numbers",
+              "type": "integer",
+              "default": 65535
+            },
+            "fallback": {
+              "description": "The port number to use if no free ports could be found",
+              "type": "integer",
+              "default": 0
+            }
+          }
+        }
+      }
+    },
+    "generatorSwitch": {
+      "properties": {
+        "generator": { "enum": [ "switch" ] },
         "parameters": {
-          "type": "object"
+          "properties": {
+            "evaluator": {
+              "description": "The style of evaluator to use",
```

Comment:
We might want to add some text here indicating that this is an advanced configuration option. Most authors won't ever need to change this.